homeassistant/components/hue/sensor_base.py

```diff
@@ -230,26 +182,26 @@ def name(self):
     @property
     def available(self):
         """Return if sensor is available."""
-        return (
-            self.bridge.available
-            and self.bridge.authorized
-            and (self.bridge.allow_unreachable or self.sensor.config["reachable"])
+        return not self.bridge.sensor_manager.coordinator.failed_last_update and (
+            self.bridge.allow_unreachable or self.sensor.config["reachable"]
         )
 
     @property
     def swupdatestate(self):
         """Return detail of available software updates for this device."""
         return self.primary_sensor.raw.get("swupdate", {}).get("state")
 
-    async def async_maybe_update_ha_state(self):
-        """Try to update Home Assistant with current state of entity.
+    async def async_added_to_hass(self):
+        """When entity is added to hass."""
+        self.bridge.sensor_manager.coordinator.async_add_listener(
+            self.async_write_ha_state  # pylint: disable=no-member
+        )
 
-        But if it's not been added to hass yet, then don't throw an error.
-        """
-        try:
-            await self._async_update_ha_state()
-        except (RuntimeError, NoEntitySpecifiedError):
-            _LOGGER.debug("Hue sensor update requested before it has been added.")
+    async def async_will_remove_from_hass(self):
+        """When entity will be removed from hass."""
+        self.bridge.sensor_manager.coordinator.async_remove_listener(
+            self.async_write_ha_state  # pylint: disable=no-member
```

Comment:
Remove pylint disable.

Reply:
Can't because it doesn't inherit from Entity.

Reply:
We should.